src/main/resources/db/changesets/20231129_snapshotrequests.yaml

```diff
@@ -0,0 +1,60 @@
+databaseChangeLog:
+  - changeSet:
+      id: snapshot_requests
+      author: rjohanek
+      changes:
+        - createTable:
+            tableName: snapshot_requests
+            columns:
+              - column:
+                  name: id
+                  type: ${uuid_type}
+                  defaultValueComputed: ${uuid_function}
+                  constraints:
+                    primaryKey: true
+                    nullable: false
+              - column:
+                  name: dataset_id
+                  type: ${uuid_type}
+                  constraints:
+                    nullable: false
+                    foreignKeyName: fk_dataset_snapshot_requests
+                    references: dataset(id)
+                    deleteCascade: true
+              - column:
+                  name: request_name
+                  type: text
+                  constraints:
+                    nullable: false
+              - column:
+                  name: request_research_purpose
+                  type: text
+                  constraints:
+                    nullable: false
+              - column:
+                  name: snapshot_builder_request
+                  type: jsonb
+                  constraints:
+                    nullable: false
+              - column:
+                  name: user_email
+                  type: text
+                  constraints:
+                    nullable: false
+              - column:
+                  name: create_date
+                  type: timestamptz
+                  defaultValueComputed: now()
+                  constraints:
+                    nullable: false
+              - column:
+                  name: update_date
+                  type: timestamptz
+                  constraints:
+                    nullable: true
+              - column:
+                  name: status
+                  type: text
+                  defaultValue: 'SUBMITTED'
+                  constraints:
```

Comment:
I was hoping that there was a way to set an enum constraint on a field but it looks like liquibase `addCheckConstraint` is a pro-only feature. Maybe not worth it for now, we're the only ones reading/writing this data. If we did want to do it, we could use a `sql` change to add the constraint manually. https://stackoverflow.com/a/5134947

Reply:
Rae and I talked about this last week, I agree that this isn't worth it for now (existing enums stored in the DB get this same treatment, and undergo serialization out of the DB to the parent enum defined in OpenAPI).